### PR TITLE
Added `on.create` Trigger

### PR DESCRIPTION
**Pull request by CodeGat**
In this PR:
- Added `on.create` trigger from https://github.com/ACCESS-NRI/model-configs-template/commit/a1988f076af334d23da3e903b490e6a24f0421bf (from ACCESS-NRI/model-configs-template#3)

NOTE: We only have to add this CI to `main` (and not other `dev-*`/`release-*` branches) because only future configurations (which are based on `ci.yml` from `main`) will require this feature. 

References ACCESS-NRI/model-config-tests#58

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -44,7 +49,7 @@ jobs:
 
   bump-tag:
     name: Tag Bump
-    if: github.event_name == 'push'
+    if: github.event_name == 'push' || (github.event_name == 'create' && github.ref_type == 'branch' && startsWith(github.ref_name, 'release-')) && github.repository != 'ACCESS-NRI/model-configs-template'
```
Comment (jo-basevi) [inline .github/workflows/ci.yml:52]: I have re-learnt that `&&` has higher precedence than `||` so I think the above is evaluated as:
```
A || (B && C)
```
In the above case should it be the following?
```
(A || B) && C
```

Comment (jo-basevi) [reply .github/workflows/ci.yml:52]: Or is evaluated left to right in this case?

Comment (CodeGat) [reply .github/workflows/ci.yml:52]: So do you mean change:
```
push || (create && branch && release) && ! template
```
to
```
(push || create) && branch && release && ! template
```
essentially?

Comment (jo-basevi) [reply .github/workflows/ci.yml:52]: Oops I meant `(push || (create && branch && release)) && ! template`? I am assuming we never want to run the step if it's the template repository

Comment (CodeGat) [reply .github/workflows/ci.yml]: You are correct!! Thank you for checking my logic, I missed that completely.

**Review by jo-basevi — APPROVED** [review]
Looks good!

**Pull request merged**